### PR TITLE
ttxdb derivates: failed restart #402

### PR DESCRIPTION
**Pull request by adecaro**
This PR fixes the `owner` and `auditor` services. Before, these services were not able to recover from a shutdown of the node. Now, at bootstrap, the services first check if there are transactions whose status is still pending and for those a commit listener is added. 

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

**Comment by adecaro** [comment]
depends on https://github.com/hyperledger-labs/fabric-smart-client/pull/409

File: token/services/auditor/manager.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		return nil, errors.Errorf("failed to get network instance for [%s:%s]", w.TMS().Network(), w.TMS().Channel())
+	}
+	logger.Debugf("auditdb: register tx status listener for all tx at network [%s:%s]", w.TMS().Network(), w.TMS().Channel())
+	return auditor, nil
+}
+
+func (cm *Manager) list() ([]Entry, error) {
+	it, err := kvs.GetService(cm.sp).GetByPartialCompositeID("auditor", nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to db list iterator")
```
Comment (KElkhiyaoui) [inline token/services/auditor/manager.go:109]: failed to get db list iterator

File: token/services/owner/manager.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+	if net == nil {
+		return nil, errors.Errorf("failed to get network instance for [%s:%s]", tms.ID().Network, tms.ID().Channel)
+	}
+	return owner, nil
+}
+
+func (cm *Manager) list() ([]Entry, error) {
+	it, err := kvs.GetService(cm.sp).GetByPartialCompositeID("owner", nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to db list iterator")
```
Comment (KElkhiyaoui) [inline token/services/owner/manager.go:102]: failed to get db list iterator

File: token/services/owner/manager.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+func New(sp view.ServiceProvider, tms *token.ManagementService) *Owner {
+	return Get(sp, tms)
+}
+
+type tmsWallet struct {
+	tms *token.ManagementService
+}
+
+func (t *tmsWallet) ID() string {
+	return ""
```
Comment (KElkhiyaoui) [inline token/services/owner/manager.go:240]: would tmsWallet always have empty string as ID?

Comment (adecaro) [reply token/services/owner/manager.go:240]: good point, let's set it to the TMS's id

**Pull request merged**